internal/key/key_test.go

```diff
@@ -116,3 +118,21 @@ func TestNewKey(t *testing.T) {
 		}
 	}
 }
+
+func TestSaveNewKey(t *testing.T) {
+	t.Parallel()
+
+	assert := assert.New(t)
+	k, err := NewSoft(0)
+	assert.NoError(err)
+	tmpFile, err := os.CreateTemp("", "save-new-key-test-file")
```

Comment:
Can we use `t.TempDir()` to generate the directory here? I think this is a good practice so the test can delete it on completion and it also catches a few bugs on different OSs if we incorrectly hold on to an fd to it.